kv/txn_coord_sender_test.go

```diff
@@ -535,10 +536,9 @@ func TestTxnCoordSenderTxnUpdatedOnError(t *testing.T) {
 		},
 		{
 			// On retry, restart with new epoch, timestamp and priority.
-			pErr: roachpb.NewError(&roachpb.TransactionRetryError{
-				Txn: roachpb.Transaction{
-					Timestamp: origTS.Add(10, 10), Priority: int32(10)},
-			}),
+			pErr: roachpb.NewError(&roachpb.TransactionRetryError{}),
+			pErrTxn: &roachpb.Transaction{
```

Comment:
The change was necessary since `TransactionRetryError` no longer has the `Txn` field. 

Updated the code to avoid adding `pErrTxn`. Instead added `NewErrorWithTxn(err, txn)` so that we can set a `pErr.Txn` easily.